src/main/java/com/google/devtools/build/lib/query2/cquery/CqueryThreadsafeCallback.java

```diff
@@ -88,6 +94,14 @@ public void close(boolean failFast) throws InterruptedException, IOException {
     }
   }
 
+  protected BuildConfiguration getConfiguration(BuildConfigurationValue.Key configKey) {
+    // Experiments querying:
```

Comment:
Use ConcurrentHashMap?

This class is named CqueryThreadsafeCallback and it extends ThreadSafeOutputFormatterCallback, so it's best to either (A) make this method thread safe (even if it's the case that cquery output formatting *currently* occurs on a single thread) or (B) make the property in the previous parenthetical explicit.

Reply:
Good point, thanks.